### PR TITLE
chore: bump go and k8s version to fix the DOS Vulnerability

### LINKED ISSUE #505: update net/http version to fix the DOS Vulnerability
```yaml
Vulnerability #1: GO-2024-2963
    Denial of service due to improper 100-continue handling in net/http
  More info: https://pkg.go.dev/vuln/GO-2024-2963
  Standard library
    Found in: net/http@go1.22.4
    Fixed in: net/http@go1.22.5
    Example traces found:
Error:       #1: pkg/builder/kaniko/kaniko.go:128:24: kaniko.Kaniko.containerLogs calls rest.Request.DoRaw, which eventually calls http.Client.Do
Error:       #2: pkg/minio/minio.go:116:31: minio.Minio.Delete calls minio.Client.StatObject, which eventually calls http.Transport.CloseIdleConnections
Error:       #3: pkg/minio/minio.go:116:31: minio.Minio.Delete calls minio.Client.StatObject, which eventually calls http.Transport.RoundTrip
```

ref: https://github.com/celestiaorg/knuu/actions/runs/9934310225/job/27438427967#step:2:9729

### PR DESCRIPTION
**Pull request by mojtaba-esk**
Closes #505

### CODE REVIEW

**Comment by mojtaba-esk** [comment]
Converted to Draft as the latest version of minio-go and k8s go client, have not updated the `http/net` version yet. We need to wait for that. Also this vulnerability has a low risk on knuu.

File: go.mod [hunk 1]
```diff
@@ -1,19 +1,18 @@
...
 	github.com/minio/minio-go/v7 v7.0.73
 	github.com/sirupsen/logrus v1.9.3
 	github.com/stretchr/testify v1.9.0
 	gopkg.in/yaml.v3 v3.0.1
-	k8s.io/api v0.28.2
-	k8s.io/apimachinery v0.28.2
-	k8s.io/client-go v0.28.2
+	k8s.io/api v0.30.2
+	k8s.io/apimachinery v0.30.2
+	k8s.io/client-go v0.30.2
```
Comment (smuu) [inline go.mod:15]: This will break `chaos-mesh`.
But as it is not in use at the moment, we should be fine doing that.

**Review by smuu — APPROVED** [review]

**Review by smuu — COMMENTED** [review]
Is it on purpose to not merge this into main?

**Comment by mojtaba-esk** [comment]
> Is it on purpose to not merge this into main?

Since the SantizeName is not yet removed on main, the tests would panic if we merge it into the main. So it is merged into another branch that has that SanitizeName fixed

**Pull request merged**